### PR TITLE
[Fix] use prop-types package + replace React.createClass

### PR DESCRIPTION
**Pull request by joeduncan**
- Use `prop-types` package instead of `React.PropTypes`
- Replace `React.createClass` with `create-react-class`

### CODE REVIEW

File: test/slider-test.jsx [hunk 1]
```diff
@@ -41,11 +41,8 @@ describeWithDOM('<Slider />', () => {
 
     it('renders pits if they are provided', () => {
       const pitRender = sinon.stub().returns(<div />);
-
-      // eslint-disable-next-line
-      const PitComponent = React.createClass({
```
Comment (ljharb) [inline test/slider-test.jsx]: We need to continue to test `React.createClass`; please use the `create-react-class` package as a dev dep instead.

Comment (joeduncan) [inline test/slider-test.jsx]: done

**Comment by ljharb** [comment]
@joeduncan this will be good to go after a command-line rebase

**Comment by joeduncan** [comment]
rebased and ready

**Pull request merged**